src/Propel/Generator/Builder/Om/AbstractOMBuilder.php

```diff
@@ -361,21 +361,23 @@ public function declareClassNamespace($class, $namespace = '', $alias = false)
      * check if the current $class need an alias or if the class could be used with a shortname without conflict
      *
      * @param string $class
-     * @param string $namespace
+     * @param string $classNamespace
      *
      * @return bool
      */
-    protected function needAliasForClassName($class, $namespace)
+    protected function needAliasForClassName($class, $classNamespace)
     {
-        if ($namespace == $this->getNamespace()) {
+        $builderNamespace = trim($this->getNamespace(), '\\');
+
+        if ($classNamespace == $builderNamespace) {
```

Comment:
not a blocker, but would be nice to have strict comparison where we can

Reply:
This is how it was initially, my guess is that it is on purpose, so that empty strings and null match.

Reply:
According to docblock null is not a valid input. But trim() could maybe return false in some cases.
I guess we need to keep this as is for now as we are not aware of possible side effects.